lake/root.go

```diff
@@ -19,14 +21,17 @@ import (
 	"github.com/segmentio/ksuid"
 )
 
+var ErrPoolExists = errors.New("pool already exists")
+var ErrPoolNotFound = errors.New("pool not found")
+
 // The Root of the lake represents the path prefix and configuration state
 // for all of the data pools in the lake.  XXX For now, we are storing the
 // pool configs in a json file without concurrency control.  We should
 // make this use a journal and check for update conflicts.
 type Root struct {
 	path     iosrc.URI
 	poolPath iosrc.URI
-	// XXX Need local mutex on config
+	configMu sync.Mutex
 	Config
```

Comment:
A public field with a private mutex is a confusing setup, but we can fix this later.

Reply:
This is temporary hack until @mccanne does the journal anyways.